src/graph/graph.c

```diff
@@ -987,32 +965,22 @@ uint64_t Graph_GetNodeDegree
 
 	// for each relationship type to consider
 	for(edgeType = start_rel; edgeType < end_rel; edgeType++) {
-		M = Graph_GetRelationMatrix(g, edgeType, false);
+		M = Graph_GetSourceRelationMatrix(g, edgeType);
 
 		//----------------------------------------------------------------------
 		// outgoing edges
 		//----------------------------------------------------------------------
 
-		// TODO: revisit once we get rid of MULTI-EDGE hack
 		if(outgoing) {
 			// construct an iterator to traverse over the source node row,
 			// containing all outgoing edges
-			RG_MatrixTupleIter_AttachRange(&it, M, srcID, srcID);
+			Delta_MatrixTupleIter_AttachRange(&it, M, srcID, srcID);
```

Comment:
Another approach would be to extract the current row into a vector and compute:
`edge_count += nvals` where `nvals` is the number of active entries in the extracted row.